DataRepo/views.py

```
@@ -569,51 +586,94 @@ def paging_form_valid(self, form):
             else:
                 order_dir = None
 
+            show_stats = False
+            if "show_stats" in cform and cform["show_stats"]:
+                show_stats = True
+
+            # Retrieve stats if present - It's ok if there's none
+            try:
+                received_stats = json.loads(cform["stats"])
+                # Apparently this causes a TypeError exception in test_views. Could not figure out why, so...
+            except (TypeError, KeyError):
+                try:
+                    if "stats" in cform:
+                        received_stats = cform["stats"]
+                    else:
+                        received_stats = None
+                except Exception as e:
```

NB: Generally, it's best to only catch a specific exception type. Perhaps we want to just let these exceptions go so we are aware of the issues, unless there are cases where we know Exceptions will be raised, but should be ignored.

---

I'd sort of added this as a debug because I was getting exceptions during development that seemed unrelated to the paging form, but since I discovered they were coming from the paging form, I'd decided it was informative to mark the issue as at least being related to the paging form, so I kept it.  The issue causing the exception was resolved.  I could raise the exception here (and on line 617) instead of log it as a warning, because there's no exception I currently expect to happen...